### DISCUSSION TITLE
`404.html` is ignored

### DISCUSSION THREAD

**Discussion by robingenz** [vercel/community#3435] [Help] [answered]
### Summary

I am using a Static Site Generator that generates a `404.html` file among other things. 
According to the [documentation](https://vercel.com/guides/custom-404-page), this file should be automatically delivered as a 404 page by Vercel if the route does not exist:
> Emit a 404.html file to your Output Directory
 and it will be served as the 404 page when a route does not match any other static file.

However, Vercel does not deliver the `404.html` file as the 404 page.
Example:
- This is my 404 page: https://capawesome.io/404.html
- But this page is actually shown as 404 page: https://capawesome.io/abc

Am I missing something?

### Example

https://capawesome.io/

### Steps to Reproduce

Deploye a static website with a `404.html` file and this `vercel.json`:

```json
{
  "version": 2,
  "public": false,
  "github": {
    "enabled": false
  },
  "builds": [{ "src": "**/*", "use": "@vercel/static" }],
  "rewrites": [
    {
      "source": "/stats/:match*",
      "destination": "https://analytics.umami.is/:match*"
    }
  ]
}
```

![grafik](https://github.com/vercel/community/assets/13857929/70c45ba7-5e72-4a2f-92b1-d013873f47ae)

**Comment by amyegan** [comment, marked as answer]
Hey @robingenz. What framework are you using to generate the site?

There are some that take over client-side routing and require custom configuration.

**Reply by robingenz** [reply]
Hey @amyegan, thanks for your fast response! I am using  [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/). It's really just a static page, not a SPA.

However, I noticed the following: It works as soon as I remove the `builds` configuration from the vercel.json file:

```diff
{
  "version": 2,
  "public": false,
  "github": {
    "enabled": false
  },
-  "builds": [{ "src": "**/*", "use": "@vercel/static" }],
  "rewrites": [
    {
      "source": "/stats/:match*",
      "destination": "https://analytics.umami.is/:match*"
    }
  ]
}
``` 

However, I do not understand what is wrong with this configuration. Is this possibly a bug on your side?

**Reply by amyegan** [reply]
I'm glad you found a workaround! The [`builds`](https://vercel.com/docs/concepts/projects/project-configuration#builds) property is not needed if your project is working correctly without it. 

That property is mainly used for unusual use cases and is not recommended for most projects. If additional configuration is still needed, you should look at converting it to [`functions`](https://vercel.com/docs/concepts/projects/project-configuration#functions) configuration instead.
